documentation/modules/cruise-control/proc-implementing-rebalance-proposal.adoc

```diff
@@ -0,0 +1,73 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='proc-implementing-optimization-proposal-{context}']
+
+= Implementing an optimization proposal
+
+You can implement an optimization proposal with a status of `ProposalReady`.
+Cruise Control will then reassign partitions to brokers as summarized in the optimization proposal.
+
+WARNING: This is not a dry run. 
+Review the optimization proposal before you implement it.
+
+.Prerequisites
+
+* You have xref:proc-generating-optimization-proposals-{context}[generated an optimization proposal].
+
+* The optimization proposal is ready to implement (the `KafkaRebalance` custom resource status is `ProposalReady`).
+
+.Procedure
+
+Perform these steps for the optimization proposal that you want to implement:
+
+. If the `KafkaRebalance` status has been `ProposalReady` for some time, refresh the optimization proposal in case it has become out of date:
+
+.. Annotate the `KafkaRebalance` resource in Kubernetes:
++
+[source,shell,subs="+quotes"]
+----
+kubectl annotate kafkarebalance _cluster-name_-kafka strimzi.io/rebalance=refresh
+----
+
+.. Check the status of the `KafkaRebalance` resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl get kafka kafkarebalance -o jsonpath='{.status}'
+----
+
+.. Wait until the status changes to `ProposalReady`.
+
+. Approve the optimization proposal that you want to implement.
++
+Annotate the `KafkaRebalance` resource in Kubernetes:
++
+[source,shell,subs="+quotes"]
+----
+kubectl annotate kafkarebalance _cluster-name_-kafka strimzi.io/rebalance=approved
+----
+
+. The Cluster Operator detects the annotated resource and instructs Cruise Control to rebalance the Kafka cluster.
+
+. Check the status of the `KafkaRebalance` resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl get kafka kafkarebalance -o jsonpath='{.status}'
+----
+
+. Cruise Control returns one of three statuses:
+
+** Rebalancing: The cluster rebalance operation is in progress. 
+
+** Ready: The cluster rebalancing operation completed successfully.
+
+** NotReady: An error occurred. 
+
+//Is there a way to verify the rebalance?
```

Comment:
Well you can ask for a new optimization proposal which should contain little or no changes? There is no direct way to check though.

Reply:
I think we should also explain how to "fix" a `KafkaRebalance` when an error occurs and how to move it from `NotReady` to ask for a new proposal.

Reply:
OK. I'd prefer to document the "fix" procedure in a separate PR after this one.